### DISCUSSION TITLE
[Proposal] Add == operator with multi check

### DISCUSSION THREAD

**Discussion by DmitrijOkeanij** [dotnet/csharplang#2947] [Language Ideas] [answered]
For now i need to write

```
if( mySomeVarable == 5 || mySomeVarable == 12 || mySomeVarable == 101 )
```

Very usefull will be short syntax

```
if(  mySomeVarable == 5 or 12 or 101 )
```

Or something like that to make quick check if variable is on of the choice.

**Comment by YairHalberstadt** [comment]
To be consistent with +=, *=, ??= etc. ` x ||= y` would have to mean `x = x || y` so a different operator would be needed.

That being said, if or patterns are added they would solve this problem for constant s: `if(mySomeVarable is 5 or 12 or 101)`

**Comment by DmitrijOkeanij** [comment]
Yes. 
May be my variant with ||= is not the best.
It can be something else, but want to solve this long expressions and made its more easy.

May be just
```
mySomeVarable == 5 or 12 or 101
```

**Comment by DmitrijOkeanij** [comment]
I rewrite upper text.
Thanks!

**Comment by theunrepentantgeek** [comment]
Have you considered that this [can be achieved](https://sharplab.io/#v2:D4AQTAjAsAUCAMACEEB0AZAlgOwI4G5ZYQBmZCANmTEQElsAXAUQA8GBTbAZ0wHtvEAb1iJRyMiioAjXrwA2iJrgCuAQzlcAgtgCeAeQBmACgYALTF0Q4GiAG7rl7ADSIADqoBOqgLaXrAbQBdOwd2LgBKETFogF4APhC5Ry5UbR0jTER4qyyYxMdwwhgAXyI4CRoAEXZvXkRYYRho0mQAFkQAWVUcDMZ89kimsUbo6MwDRCN7JPZUJTUNNMMjAFYXCDB1+Ahw8Pqh0aEow+iAelPEdjlMABN2G+PR0oPEZ+KgA=) (for your specific example) with an extension method?

``` csharp
using System;
using System.Linq;

public static class IntExtensions {
    public static bool EqualsAnyOf(this int value, params int[] values)
        => values.Any(i => i == value);
}

public class Demo 
{
    public void Main(int value)
    {
        if (value.EqualsAnyOf(5, 12, 101)) 
        {
            // elided
        }
    }
}
```

More generally, you should check out #2850 (_Proposed changes for Pattern Matching in C# 9.0 - Draft Specification_) as the syntax proposed there will do exactly what you are requesting.

**Comment by spydacarnage** [comment]
And you can make that a generic method, not just for int, and it covers all
bases.

**Comment by TonyValenti** [comment]
I would really prefer to see an “in” operator that was essentially an alias for “contains”.

If(x in {1, 2, 3} ) {
...
}

**Comment by Richiban** [comment, marked as answer]
This is likely to be implemented with "and, or, and not patterns" https://github.com/dotnet/csharplang/issues/1350.

```csharp
if (x is 1 or 2 or 3)
{
    //...
}
```
